### ISSUE TITLE
Subscription issue

### ISSUE THREAD

**Issue by revskill10** [prisma/prisma-binding#229]
I'm following tutorial at howtographql.com, here's my subscription for `newLink`:
```js
function newLinkSubscribe (parent, args, context, info) {
  return context.db.subscription.link(
    { where: { mutation_in: ['CREATED'] } },
    info,
  )
}

const newLink = {
  subscribe: newLinkSubscribe
}

module.exports = {
  newLink,
}
```
My subscription query is:
```graphql
subscription RealTimeFeed {
    newLink {
      mutation
      node{
        url
      }
    }
  }
```
But when i post a mutation to create new link
```graphql
mutation {
  post(
    url: "fdsfdsdsdf.graphqlweesfsdfkly.com"
    description: "Curated GraphQL content coming to your email inbox every Friday"
  ) {
    id
  }
}
```

, i got error : 
```json
{
  "error": {
    "name": "TypeError",
    "message": "Cannot read property 'link' of undefined"
  }
}
```
But if i go to database, schema, i got correct subscription:
```json
{
  "data": {
    "link": {
      "mutation": "CREATED",
      "node": {
        "url": "fdsfdsdsdf.graphqlweesfsdfkly.com"
      }
    }
  }
}
```

This is killing me hardly. Any magic for subscription only work for database, but not on `app` ?.

My `schema.graphql` is:
```graphql
# import Link, LinkSubscriptionPayload  from "./src/generated/prisma.graphql"

type Query {
  info: String!
  feed: [Link!]!

  # Fetch a single link by its `id`
  link(id: ID!): Link
  user: User
}

type Mutation {
  post(url: String!, description: String!): Link!
  signup(email: String!, password: String!, name: String!): AuthPayload
  signinUser(email: String!, password: String!): AuthPayload
}

type AuthPayload {
  token: String
  user: User
}

type User {
  id: ID!
  name: String!
  email: String!
  links: [Link!]!
}

type Subscription {
  newLink: LinkSubscriptionPayload
}
```
My `package.json`:
```json
{
  "name": "hackernews-node",
  "version": "1.0.0",
  "main": "index.js",
  "license": "MIT",
  "dependencies": {
    "bcryptjs": "^2.4.3",
    "graphql-yoga": "^1.16.0",
    "jsonwebtoken": "^8.3.0",
    "prisma-binding": "^2.1.4"
  }
}
```

**Comment by joefru** [comment]
Had the same issue when upgrading prisma-binding to 2.1.4. I suspect a bug there. When I **downgrade to prisma-binding@2.1.3** the problem goes away.

**Comment by revskill10** [comment]
@joefru I got same problem even with `2.1.3`, weird.

**Comment by maury91** [comment]
Related: https://github.com/howtographql/howtographql/issues/758

**Comment by maury91** [comment]
Same for me, downgrading to `2.1.3` solved the issue

**Comment by maury91** [comment]
The error is in this line:

https://github.com/graphql-binding/graphql-binding/blob/master/src/Delegate.ts#L88

I changed from

```
const data = { [info.fieldName]: value.data[fieldName] }
```

to

```
const data = { [info.fieldName]: value[fieldName] }
```

and it solved the error, but this raises the question. Should the value pass to Delegate contain a field named data?

Because if the answer is yes, then the error is on the part of the code where we are not wrapping that object in ".data"

**Comment by maury91** [comment]
The issue still exists, leave it open until is fixed in the next version

**Comment by revskill10** [comment]
@maury91 I reopened the issue for who's interested. There're some serious issues here and there. 
Basically, the only way to make all these issues less hurt for developers to upgrade devs, is to rewrite the tutorial following TDD, make library for test so that we could write clean test before any coding.
It saves time as well as give more helpful debugging stories to fix them.

**Comment by joealden** [comment]
@schickling @timsuchanek this issue is blocking me from updating. Would love to see it fixed.

**Comment by ntziolis** [comment]
Same here, in fact it cost me about 8h yesterday while I was debugging using fragments with subscriptions (which then resulted in another bug raised). I agree with other devs that the continuous introductions of major bugs like non working subscriptions is something to worry about. Can we assist you in any way setting up better regression testing?

**Comment by ntziolis** [comment]
@maury91 I can confirm that making the changes suggested fixes the issue.

**Comment by maury91** [comment]
@ntziolis  @SilencerWeb  the main problem is that the issue is on another repository ( https://github.com/graphql-binding/graphql-binding ), and we don't know if consumers that are not prisma-binding are affected, by making that change we can potentially break another consumer, so is worth first investigating if that is the expected behaviour and is prisma-binding the one passing the wrong parameters to Delegate, or if it is the opposite and proceed with the fix.

**Comment by ntziolis** [comment]
I see, lmk if I can assist in any way. We need this current version of the bindings as its the first one to allow fragments in subscriptions. Have been waiting for that fix for a long time as it will significantly clean up our code based and reduce errors of devs by not keeping fragment in subscription in sync.

**Comment by etx** [comment]
Same problem here. 2.1.3 my subs work but when moving to 2.1.4 they error out

`(node:4867) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'section' of undefined
    at Object.<anonymous> (/path/node_modules/graphql-binding/dist/Delegate.js:88:105)
    at step (/path/node_modules/graphql-binding/dist/Delegate.js:32:23)
    at Object.next (/path/node_modules/graphql-binding/dist/Delegate.js:13:53)
    at fulfilled (/path/node_modules/graphql-binding/dist/Delegate.js:4:58)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:160:7)
(node:4867) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwinginside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 2)
(node:4867) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.`

**Comment by simonjoom** [comment]
Maury91 well found the issue of this problem:
https://github.com/prisma/prisma-binding/issues/229#issuecomment-413913264

world for me

**Comment by divyenduz** [comment]
This is fixed in `2.2.4` 🎉

**Comment by akashanand708** [comment]
downgrading to 2.1.3 solved the issue
